apps/src/maze/maze.js

```diff
@@ -1170,9 +1168,14 @@ Maze.scheduleAnimations = function (singleStep) {
     }
 
     animateAction(actions[index], singleStep, timePerAction);
+
+    var command = actions[index] && actions[index].command;
+    var timeModifier = (skin.actionSpeedScale && skin.actionSpeedScale[command]) || 1;
+    var timeForThisAction = parseInt(timePerAction / timeModifier, 10);
```

Comment:
Ahhh, I wasn't considering that. Might be good to pass it an int (though I might Math.round or something instead of parseInt if we're just converting float to int)